### DISCUSSION TITLE
A proper Java API library doesn't exist

### DISCUSSION THREAD

**Discussion by Haarolean** [octokit/.github#42] [Ideas]
It feels really discouraging not to have a proper GitHub API client written in Java, which is one of the most popular languages out there.

There are a few things in existence by today:

1. [hub4j/github-api](https://github.com/hub4j/github-api) -- not really maintained, at least half of events/hooks are not implemented, and the code is a mess.
2. [jcabi/jcabi-github](https://github.com/jcabi/jcabi-github) -- Github Apps API is not supported, webhooks are not supported, and very poor functionally.
3. Tried building [github/rest-api-description](https://github.com/github/rest-api-description) with openapi-generator, there is a ton of bugs that prevent proper compiling, rendering this thing unviable.

It would be nice to have a proper Java client with official maintenance, it would allow developers familiar with Java to extend all the possibilities Github API provides.

**Comment by kfcampbell** [comment]
Hi! I work on the GitHub SDK team and this is something we're thinking about! In our perfect world, we'd have generated SDKs for several major languages including Java. In the near future, we will not be creating a hand-maintained Java SDK similar to octokit.net for example, but we are working hard to make that generative future a reality. When it's ready, that work will be published under the Octokit organization.

**Reply by gungwald** [reply]
It's thee years later. Is there any update? I'm using hub4j/github-api. It is hard to make progress with.

**Reply by wolfy1339** [reply]
Unfortunately, GitHub isn't devoting any resources to Octokit anymore.
